processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -257,9 +249,23 @@ func (c *WatchClient) Start() error {
 			return err
 		}
 		synced = append(synced, reg.HasSynced)
-		go c.nodeInformer.Run(c.stopCh)
+		waitForInformers = append(waitForInformers, c.nodeInformer)
+		go c.runInformer(c.nodeInformer)
 	}
 
+	reg, err = c.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.handlePodAdd,
+		UpdateFunc: c.handlePodUpdate,
+		DeleteFunc: c.handlePodDelete,
+	})
+	if err != nil {
+		return err
+	}
+	synced = append(synced, reg.HasSynced)
+
+	// start the podInformer with the prerequisite of the other informers to be finished first
+	go c.runInformer(c.informer, waitForInformers...)
```

Comment:
If we pass the `synced` array of the dependency informers here we are saved from passing the complete Informers' objects to the function.